addons/project_timesheet_holidays/models/hr_holidays.py

```diff
@@ -135,7 +135,16 @@ def action_refuse(self):
     def _action_user_cancel(self, reason):
         res = super()._action_user_cancel(reason)
         timesheets = self.sudo().timesheet_ids
-        timesheets.write({'holiday_id': False})
+        timesheets.holiday_id = False
         timesheets.unlink()
         self._check_missing_global_leave_timesheets()
         return res
+
+    def _force_cancel(self, *args, **kwargs):
+        """
+        override this method to reevaluate timesheets after the leaves are updated via force cancel
+        """
+        super()._force_cancel(*args, **kwargs)
+        timesheets = self.sudo().timesheet_ids
+        timesheets.holiday_id = False
```

Comment:
Is this line useful, since we unlink the timesheets right after? I think the default delete policy should be “set null” in this case

Reply:
This line was actually necessary for this task, and removing this line raises an error: `You cannot delete timesheets linked to time off. Please, cancel the time off instead`